mypy/checker.py

```diff
@@ -4538,6 +4538,21 @@ def named_type(self, name: str) -> Instance:
         any_type = AnyType(TypeOfAny.from_omitted_generics)
         return Instance(node, [any_type] * len(node.defn.type_vars))
 
+    def named_type_or_none(self, qualified_name: str,
+                           args: Optional[List[Type]] = None) -> Optional[Instance]:
+        sym = self.lookup_qualified(qualified_name)
+        if not sym or isinstance(sym.node, PlaceholderNode):
```

Comment:
No need to check for placeholders, they should never appear at this stage.

Reply:
Removed the check